src/cockpit/channels/filesystem.py

```diff
@@ -166,18 +167,40 @@ class FsWatchChannel(Channel):
     # Ideally we'll sort that out at some point, but for now, suppress it.
     _active = False
 
-    def set_tag(self, tag):
-        if tag == self._tag:
-            return
-        self._tag = tag
-        if self._active:
-            self.send_message(path=self._path, tag=self._tag)
-
-    def do_inotify_event(self, _mask, _cookie, _name):
-        self.set_tag(tag_from_path(self._path))
+    @staticmethod
+    def mask_to_event_and_type(mask):
+        if mask & (InotifyEvent.CREATE | InotifyEvent.MOVED_TO):
```

Comment:
Urgh, I'd find that pretty obfuscated for a bitwise operation, it would look better for a set. Also, it doesn't seem to work?
```
>>> mask = 5
>>> 1 in mask
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: argument of type 'int' is not iterable
>>> (1 or 5) in mask
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: argument of type 'int' is not iterable
```

Reply:
`5` is an `int`.  `InotifyEvent` is a [`enum.IntFlag`](https://docs.python.org/3/library/enum.html#enum.IntFlag).

Reply:
fwiw, the "int-like" behaviour of IntFlag is officially only there for compatibility reasons.

Reply:
Unless you insist, I'd rather keep it as bitwise operation. I find that much more obvious what happens (maybe I'm old fashioned).

Reply:
Discussed on IRC, will change. I just found one gotcha:

    (InotifyEvent.MOVED_FROM or InotifyEvent.DELETE or InotifyEvent.DELETE_SELF) in mask

This does not work, only a single `or` does. So I rewrote it accordingly.

Reply:
That's pretty awful :(  Sorry.

Reply:
@allisonkarlitskaya Do you still care about the above? :arrow_up: That's still on my TODO list.